code/game/objects/items/devices/PDA/PDA.dm

```diff
@@ -978,17 +984,17 @@ var/global/list/obj/item/device/pda/PDAs = list()
 		return
 
 	last_text = world.time
-	var/datum/reception/reception = get_reception(src, P, t)
-	t = reception.message
 
-	if(reception.message_server && (reception.telecomms_reception & TELECOMMS_RECEPTION_SENDER)) // only send the message if it's stable
-		if(reception.telecomms_reception & TELECOMMS_RECEPTION_RECEIVER == 0) // Does our recipient have a broadcaster on their level?
+	if(src.has_connection()) // only send the message if it's stable
+		if(!P.has_connection()) // Does our recipient have a broadcaster on their level?
 			U << "ERROR: Cannot reach recipient."
 			return
-		var/send_result = reception.message_server.send_pda_message("[P.owner]","[owner]","[t]")
-		if (send_result)
-			U << "ERROR: Messaging server rejected your message. Reason: contains '[send_result]'."
-			return
+
+		for(var/obj/machinery/message_server/M in message_servers)
+			var/send_result = M.send_pda_message("[P.owner]","[owner]","[t]") // We need to make logs in all servers

```

Comment:
Hm... you won't get multiple rejection messages, because it returns - but it also means that it will only be logged in the first server.
Updates... ugh. I think I'll just up the limit and remove the upgrading.